@xen-orchestra/backups/writers/DeltaBackupWriter.js

```diff
@@ -211,6 +213,9 @@ exports.DeltaBackupWriter = class DeltaBackupWriter extends MixinBackupWriter(Ab
               nbdClient = new NbdClient(nbdInfo)
               await nbdClient.connect()
               info('NBD client ready', { vdi: id, path })
+              $defer(async () => {
+                await nbdClient.disconnect()
```

Comment:
Isn't the ideal case for a disposable?

A resource that can be used once initialized and must be disposed of afterward?

Reply:
I tried with  disposable, but I think it was less lisible, since nbdClient is optional, and I didn't want to put the logic of what to do when  there is no nbd info, or when the server is not answering in `nbd-client`

Reply:
It should be pretty easy to do with `Disposable.wrap` but it's up to you :slightly_smiling_face: